pkg/server/server.go

```diff
@@ -493,6 +493,7 @@ func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer,
 			// Currently not worrying about backend as we do not have an established connection,
 			s.PendingDial.Remove(random)
 			klog.V(5).InfoS("Removing pending dial request", "serverID", s.serverID, "dialID", random)
+			metrics.Metrics.ObserveDialFailure(metrics.DialFailureFrontendClose)
```

Comment:
Only count if there was a pending dial to remove.

Reply:
+1.